src/cpu/x64/jit_copy_1d.cpp

```diff
@@ -0,0 +1,1284 @@
+        vmovups(yword[rax + rcx] | k1, ymm2);
+        vpcmpgtd(k1 | k1, ymm0, ymm1);
+        add(eax, 32);
+        kortestb(k1, k1);
+        jne(LBB11_2);
+        L(LBB11_3); // %for_exit
+        pop(rbx);
+        vzeroupper();
+        ret();
+    }
+};
+
+template <typename T>
+void copy_1d(T *dst, const T *src, int start_off, int src_width, int nelem) {
+    static jit_copy_1d<T, 1> j;
+    if (!j.is_generic())
+        return j(dst, src, start_off, src_width, nelem);
+    else {
+        for (int i = 0; i < nelem; ++i) {
+            auto _off = i + start_off;
+            dst[i] = (_off >= 0 && _off < src_width) ? src[_off] : (T)0;
+        }
+    }
+}
+
+template <typename T>
+void copy_1d_batch_4(T *dst, int d_stride, const T *src, int s_stride,
+        int start_off, int src_width, int nelem) {
+    static jit_copy_1d<T, 4> j;
+    if (!j.is_generic())
+        return j(dst, d_stride, src, s_stride, start_off, src_width, nelem);
+    else {
+        for (int i = 0; i < nelem; ++i) {
+            auto p_off = i + start_off;
+            for (int batch = 0; batch < 4; ++batch) {
+                (dst + batch * d_stride)[i] = (p_off >= 0 && p_off < src_width)
+                        ? (src + batch * s_stride)[p_off]
+                        : (T)0;
+            }
+        }
+    }
+}
+
+template <typename T>
+void copy_1d_batch_8(T *dst, int d_stride, const T *src, int s_stride,
+        int start_off, int src_width, int nelem) {
+    static jit_copy_1d<T, 8> j;
+    if (!j.is_generic())
+        return j(dst, d_stride, src, s_stride, start_off, src_width, nelem);
+    else {
+        for (int i = 0; i < nelem; ++i) {
+            auto p_off = i + start_off;
+            for (int batch = 0; batch < 8; ++batch) {
+                (dst + batch * d_stride)[i] = (p_off >= 0 && p_off < src_width)
+                        ? (src + batch * s_stride)[p_off]
+                        : (T)0;
+            }
+        }
+    }
+}
+
+template <typename T>
+void zero_1d(T *dst, int nelem) {
+    static jit_zero_1d<T, 1> j;
+    if (!j.is_generic())
+        return j(dst, nelem);
+    else {
+        for (int pid = 0; pid < nelem; ++pid) {
+            dst[pid] = (T)0;
+        }
+    }
+}
+
+template <typename T>
+void zero_1d_batch_4(T *dst, int stride, int nelem) {
+    static jit_zero_1d<T, 4> j;
+    if (!j.is_generic())
+        return j(dst, stride, nelem);
+    else {
+        for (int pid = 0; pid < nelem; ++pid) {
+            for (int batch = 0; batch < 4; ++batch) {
+                (dst + batch * stride)[pid] = (T)0;
+            }
+        }
+    }
+}
+
+template <typename T>
+void zero_1d_batch_8(T *dst, int stride, int nelem) {
+    static jit_zero_1d<T, 8> j;
+    if (!j.is_generic())
+        return j(dst, stride, nelem);
+    else {
+        for (int pid = 0; pid < nelem; ++pid) {
+            for (int batch = 0; batch < 8; ++batch) {
+                (dst + batch * stride)[pid] = (T)0;
+            }
```

Comment:
Furthermore as I understand,  non-specialized generic code will run for `bfloat16_t` data type (where `T = uint16_t`) as well

Reply:
Another concern is about the `order` of loop operations. Should not the inner-most loop iterate over `pid` so as to benefit from cache locality.

Reply:
Okey, in this case, both nearly the same. I wrote this way for algorithm purpose and I think PRAGMA_OMP_SIMD() might have the chance to create the same kernel as ISPC.

Reply:
> Furthermore as I understand, non-specialized generic code will run for `bfloat16_t` data type (where `T = uint16_t`) as well

We don't want to introduce datatype based dispatching because we might support uint16_t fast path in the future.